packages/hierarchies-react/src/presentation-hierarchies-react/internal/TreeActions.ts

```diff
@@ -66,26 +94,12 @@ export class TreeActions {
 
   private loadSubTree(options: LoadNodesOptions, initialRootNode?: TreeModelRootNode) {
     const loadAction = this.getLoadAction(options.parent.id);
-    const timeTracker = new TimeTracker((time) => this._onLoad(loadAction, time));
-    const parentId = options.parent.id;
-    this._loader
-      .loadNodes(options)
-      .pipe(collectTreePartsUntil(this._reset, initialRootNode))
-      .subscribe({
-        next: (newModel) => {
-          const childNodes = newModel.parentChildMap.get(parentId);
-          const firstChildNode = childNodes?.length ? newModel.idToNode.get(childNodes[0]) : undefined;
-          this.handleLoadedHierarchy(parentId, newModel);
-          // only report load duration if no error occurs
-          if (!(firstChildNode && isTreeModelInfoNode(firstChildNode))) {
-            timeTracker.finish();
-          }
-        },
-        complete: () => {
-          this.onLoadingComplete(parentId);
-          timeTracker[Symbol.dispose]();
-        },
-      });
+    this._timeTracker = new TimeTracker((time) => this._onLoad(loadAction, time));
+    this._parentId = options.parent.id;
+    this._initialRootNode = initialRootNode;
+
+    this._nodeLoader.next(options);
```

Comment:
`loadSubTree` is used not just to reload the whole tree, but also when expanding nodes. It looks like the change might cause only the last requested node to expand if we call `expandNode` multiple times with different `parentId` arguments.

Reply:
I'm wondering if it's enough to group by `parentId` and just take the last request from the group. Wouldn't that cause problems in cases like the following?

```
actions.reloadTree({ parentNodeId, state: "discard" });
actions.reloadTree({ parentNodeId, state: "keep" });
```

The first call would be ignored and the state would not be discarded, right?

Reply:
The first request would be ignored, in this case do we want the state to be discarded? If we do this might be a bit tricky as the options passed to the pipe are the same (the difference is the return for shouldLoadChildren)

Reply:
Made it so if multiple reloads where made and atleast one of them was discard it would use that request.